### PR TITLE
[GHA] Changed ubuntu build runner to 20.04

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Changed the version of ubuntu to 20.04 to resolve timeout issue with latest runner (killed after 3 hours)

### Tickets:
 - *n/a*

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     defaults:
```
Comment (rkazants) [inline .github/workflows/linux.yml:32]: please change the title of workflow above:
Tests on Linux (Ubuntu 22.04, Python 3.11) -> Tests on Linux (Ubuntu 20.04, Python 3.11)

Also, please change machines for other jobs in the flow. And use ubuntu-20.04-8-cores for TensorFlow Hub Model tests

Comment (mryzhov) [reply .github/workflows/linux.yml:32]: yes, thanks, fixed

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -807,7 +807,7 @@ jobs:
...
       run:
         shell: bash
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply .github/workflows/linux.yml:810]: maybe we need to use larger runner only for nightly? 
For precommit it's cheaper to use default one

**Comment by rkazants** [comment]
@mryzhov, please pay attention to the build time. In my experiment, I observed the significant time increase up to 2h.
May be we also need to care of build caching for this new configuration.

cc @akladiev, @ilya-lavrenov

File: .github/workflows/linux.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (akashchi) [reply .github/workflows/linux.yml:810]: Yes, it will.

**Review by rkazants — APPROVED** [review]

File: .github/workflows/linux.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mryzhov) [reply .github/workflows/linux.yml]: > maybe we need to use larger runner only for nightly?
> For precommit it's cheaper to use default one

it depends on execution time, will monitor it and correct if needed

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -807,7 +807,7 @@ jobs:
     defaults:
       run:
         shell: bash
-    runs-on: ubuntu-22.04
+    runs-on: ${{ github.event_name == 'schedule' && 'ubuntu-20.04-8-cores' || 'ubuntu-20.04'}}
     env:
```
Comment (ilya-lavrenov) [inline .github/workflows/linux.yml:811]: please, update after merging this PR https://github.com/openvinotoolkit/openvino/pull/19702

Comment (rkazants) [reply .github/workflows/linux.yml:811]: Done

**Review by akashchi — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**